tket/include/tket/Predicates/CompilerPass.hpp

```diff
@@ -301,6 +299,14 @@ class RepeatUntilSatisfiedPass : public BasePass {
   PredicatePtr pred_;
 };
 
+nlohmann::json serialise(const BasePass& bp);
+nlohmann::json serialise(const PassPtr& pp);
+nlohmann::json serialise(const std::vector<PassPtr>& pp);
+
+PassPtr deserialise(
+    const nlohmann::json& j,
+    const std::map<std::string, std::function<Circuit(const Circuit&)>>&
```

Comment:
If we called these `to_json` and `from_json` as before, can we avoid some of the changes in `CompilerPass.cpp` (not sure how the magic in `nlohmann::json` works).

Reply:
(And also change the signature to match the signatures in `JSON_DECL` but with the addition of the one optional argument.)

Reply:
My look through the nlohmann docs/code suggested that this wasn't possible, with the nlohmann magic relying one using well defined to_json/from_json with two arguments - the json and the object. So I don't think this is possible